### PR TITLE
Adding capability of Sulong to parse bitcode files

### PR DESCRIPTION
**Pull request by crbb**
Allow bitcode files to be decoded using su-run (using the same LLVM version as already used by Sulong) and assemble files in SulongTestSuites to pass the same tests as su-tests-sulong.

### CODE REVIEW

File: projects/com.oracle.truffle.llvm/src/com/oracle/truffle/llvm/LLVM.java [hunk 1]
```diff
@@ -95,6 +95,15 @@ public CallTarget parse(Source code, Node contextNode, String... argumentNames)
                     if (!context.isParseOnly()) {
                         parserResult.getStaticInits().call();
                     }
+                } else if (code.getMimeType().equals(LLVMLanguage.LLVM_BITCODE_MIME_TYPE)) {
```
Comment (mrigger) [inline projects/com.oracle.truffle.llvm/src/com/oracle/truffle/llvm/LLVM.java]: There is now a method `handleParserResult` to prevent code duplication.

**Comment by mrigger** [comment]
Thanks a lot! Great to see that the parser passes (again) all the Sulong test cases!

**Comment by chrisseaton** [comment]
I don't understand the comment 'su-run (using the same LLVM version as already used by Sulong)' - are we restricted to one LLVM version?

**Comment by crbb** [comment]
The decoder currently uses the same version of LLVM as Sulong to maintain compatibility with Dragonegg. It has the ability to handle LLVM 3.8 but an option will be needed to switch between versions from the command line.

**Pull request closed without merging**